packages/core-mobile/app/hooks/earn/useGetFeeState.ts

```diff
@@ -0,0 +1,34 @@
+import { pvm } from '@avalabs/avalanchejs'
+import { useAvalancheXpProvider } from 'hooks/networks/networkProviderHooks'
+import { useCallback, useEffect, useState } from 'react'
+import Logger from 'utils/Logger'
+
+export const useGetFeeState = (): {
+  getFeeState: (gasPrice?: bigint) => pvm.FeeState | undefined
+} => {
+  const xpProvider = useAvalancheXpProvider()
+  const [feeState, setFeeState] = useState<pvm.FeeState | undefined>(undefined)
+
+  useEffect(() => {
+    const fetchFeeState = async (): Promise<void> => {
+      if (xpProvider === undefined) return
+      const fee = await xpProvider
+        .getApiP()
+        .getFeeState()
+        .catch(() => undefined)
+      setFeeState(fee)
+    }
+    fetchFeeState().catch(Logger.error)
+  }, [xpProvider])
+
+  const getFeeState = useCallback(
+    (gasPrice?: bigint) => {
+      return feeState
+        ? { ...feeState, price: gasPrice ?? feeState.price }
+        : undefined
+    },
+    [feeState]
+  )
```

Comment:
I think you should add a getDefaultFeeState to return feeState as is and another with gasPrice passed in to make it clear in the code which is which